Web/starter_project_iconics/components/profile/ProfileLayout.tsx

```diff
@@ -0,0 +1,83 @@
+import Link from 'next/link'
+import React, { useState } from 'react'
+
+type ProfileProps = {
+  buttonText: string
+  element: React.ReactNode
+  text: string
+  innerText: string
+  currentPage: string
+}
+
+const ProfileLayout: React.FC<ProfileProps> = ({
+  buttonText,
+  element,
+  text,
+  innerText,
+  currentPage,
+}) => {
+  const [activeLink, setActiveLink] = useState(currentPage)
+
+  const handleLinkClick = (linkText: string) => {
+    setActiveLink(linkText)
+  }
+
+  return (
+    <div className="min-h-screen bg-white flex p-10 flex-col">
+      <div className="flex   py-4">
+        <h1 className="text-4xl font-bold text-gray-900">Profile</h1>
+      </div>
+      <div className="flex flex-col md:flex-row justify-center items-center bg-white pb-0">
```

Comment:
You can store this in array and loop over it too. The code seems a bit redundant.